sdk/batch/Microsoft.Azure.Batch/tests/IntegrationTests/IntegrationTestUtilities/TestUtilities.cs

```diff
@@ -62,6 +65,28 @@ public static async Task<BatchClient> OpenBatchClientFromEnvironmentAsync()
             return client;
         }
 
+
+        public static async Task<string> GetAccessToken(string[] scopes)
+        {
+            var app = ConfidentialClientApplicationBuilder.Create(TestCommon.Configuration.ClientId)
+                        .WithClientSecret(TestCommon.Configuration.ClientKey)
+                        .WithAuthority(new Uri(AuthorityUri + TestCommon.Configuration.BatcTenantID))
+                        .Build();
+
+            var result = await app.AcquireTokenForClient(scopes).ExecuteAsync();
+            return result.AccessToken;
+        }
+
+        public static async Task<BatchClient> OpenBatchClientServicePrincipal()
+        {
+            var token = await GetAccessToken(new string[] { TestCommon.Configuration.BatchAccountUrl + ".default" });
```

Comment:
FYI, this should be fine for tests as long as they don't take too longe to run, but I believe for any real scenario we'd need to pass a function into the credentials which gets the token when needed rather than a hard-coded access token. This is so the token can be refreshed when it expires. (Default expiry is 60 minutes I think)

Reply:
yeah i agree, when i port this test over to track2 I'll have it pass a function in so we don't run into expired credentialss